ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -790,11 +853,12 @@ private void prepareForSending() {
         chunkBlobId =
             new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), passedInBlobProperties.getAccountId(),
                 passedInBlobProperties.getContainerId(), partitionId);
-        chunkBlobProperties = new BlobProperties(buf.remaining(), passedInBlobProperties.getServiceId(),
-            passedInBlobProperties.getOwnerId(), passedInBlobProperties.getContentType(),
-            passedInBlobProperties.isPrivate(), passedInBlobProperties.getTimeToLiveInSeconds(),
-            passedInBlobProperties.getCreationTimeInMs(), passedInBlobProperties.getAccountId(),
-            passedInBlobProperties.getContainerId(), passedInBlobProperties.isEncrypted());
+        chunkBlobProperties = new BlobProperties(passedInBlobProperties.isEncrypted() ? chunkBlobSize : buf.remaining(),
```

Comment:
I would prefer separate variables for `chunkBlobSize` and `encryptedChunkBlobSize` etc. and use them as appropriate - rather than using `chunkBlobSize` for one and `buf.remaining()` for the other.

Reply:
sure. will fix it